File: JUNGOL/Beginner/#1856 숫자사각형2/숫자사각형2.cpp
#include <iostream>
using namespace std;

int main(){
    int n,m; cin>>n>>m;

    int num=1, dir=1;
    for(int i=0; i<n; i++){
        if(dir == 1){
            for(int j=0; j<m; j++)
                printf("%d ", num++);
            cout<<endl;
            dir = -dir;
        }
        else{
            for(int j=num+m-1; j>=num; j--)
                printf("%d ", j);
            cout<<endl;
            num += m;
            dir = -dir;
        }
    }
}

File: JUNGOL/Beginner/#1856 숫자사각형2/숫자사각형2.py
n,m = map(int, input().split())

num,dir = 1,1
for i in range(n):
    if dir == 1:
        for j in range(m):
            print(num, end=' ')
            num += 1
        print()
        dir = -dir
    else:
        for j in range(num+m-1, num-1, -1):
            print(j, end=' ')
        print()
        num += m
        dir = -dir
